src/main/kotlin/org/rust/ide/inspections/RsSelfInStaticContextInspection.kt

```diff
@@ -0,0 +1,32 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.mixin.isAssocFn
+import org.rust.lang.core.psi.util.parentOfType
+
+class RsSelfInStaticContextInspection : RsLocalInspectionTool() {
+    override fun getDisplayName(): String = "self in static context"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
+        object : RsVisitor() {
+            override fun visitPath(path: RsPath) = inspectPath(holder, path)
+        }
+
+    private fun inspectPath(holder: ProblemsHolder, path: RsPath) {
+        if (path.self == null) {
+            return
+        }
+
+        val function = path.parentOfType<RsFunction>() ?: return
+
+        if (function.isAssocFn) {
+            holder.registerProblem(
+                path,
+                "The self keyword was used in a static method [E424]",
```

Comment:
It's [E0424]. And I think the real Rust message differs from this one because you actually can use the `self` keyword in a static context, just not everywhere (see the comment about a false positive).